### DISCUSSION TITLE
Override Config.h

### DISCUSSION THREAD

**Discussion by LucaCortelli** [grblHAL/core#325] [General]
hi, I would like to create a configuration for my machine that allows me to define some parameters (N_AXIS 4 , COMPATIBILITY_LEVEL 0 etc) without modifying config.h where should I define these parameters to override those in config.h?

**Comment by terjeio** [comment]
On the compiler command line (-D arguments).

**Reply by LucaCortelli** [reply]
in the arduino ide there is this option?

**Reply by terjeio** [reply]
I do not think so, it is basically limited to editing and compiling. [PlatformIO](https://github.com/grblHAL/iMXRT1062#platformio) would be a better option, it allows project specific compiler command line options in the [project .ini file](https://github.com/grblHAL/iMXRT1062/blob/master/grblHAL_Teensy4/platformio.ini). FYI I am using PlatformIO in the backend for the [Web Builder](http://svn.io-engineering.com:8080/), the build options for that are all handled via compiler command line options.

**Reply by LucaCortelli** [reply]
I'll try. thanks for the tip
